<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
  <meta http-equiv="content-type"
 content="text/html; charset=ISO-8859-1">
  <title>SEURAT: Visual analytics for the integrative analysis of
microarray data</title>
  <meta name="Martin Sill" content="Seurat">
  <meta name="keywords"
 content="Microarrays,Integrative,Interactive,Visualization">
  <link rel="stylesheet" href="style.css" type="text/css">
</head>
<body>
<div id="inhalt">
<h1>Data formats</h1>
<hr>
<br>
<br>
<h2>General</h2>
<hr>
<br>
<p align="justify">
All data files have to be in tab delimited ascii format and missing
values should be labeled "NA". Example files are available in the <a
 href="down.html" target="PageRechts">Download and Installation</a>
section.</p>
<br>
<br>
<h2>Gene expression data</h2>
<hr>
<br>
<p align="justify">SEURAT provides no preprocessing tools, gene
expression data should be
normalized and preprocessed beforehand. It is mandatory that the first
column contains the gene IDs. The following columns should be named
according to the sample IDs and contains the gene expression values
of the respective samples.
</p>
<br>
<br>
Example: <br>
<br>
<pre>geneID   sample.1 sample.2  ...   sample.n<br>gene.1   y.11     y.12     ...    y.1n<br>gene.2   y.21     y.22     ...    y.2n<br>...<br>gene.p   y.p1     y.p2     ...    y.pn<br>  </pre>
<br>
<h2>Clinical data</h2>
<hr>
<br>
<p align="justify">
The first column should contain the sample IDs and these IDs should
have the same format as in the gene expression file.&nbsp; Additional
columns can contain any type of continous or categorical variable with
the name of the variable in the first row.</p>
<br>
<br>
Example: <br>
<br>
<pre>sampleID   variable.1 variable.2  ...   variable.z<br>sample.1   x.11       x.12     ...      x.1z<br>sample.2   x.21       x.22     ...      x.2z<br>...<br>sample.n   x.n1       x.n2     ...      x.nz<br>  </pre>
<br>
<h2>Gene annotations</h2>
<hr>
<br>
<p align="justify">
The first column has to contain the geneIDs and the gene names must be
the same as the gene names in the gene expression file. In case of
pathway informations where each gene can belong to different pathways,
the entries of pathway variables have to be lists of comma delimited
strings, e.g.: "cell_signaling, immunology, metastasis" or for
GO-terms: "molecular function|ATP binding|IEA|GO:0005524|GOA/IPI,
biological process|defense response to pathogen|IEA|GO:0042829|GOA/IPI,
cellular component|nucleus|IEA|GO:0005634" The name of a pathway
variable has to begin with "#" as the first character.<br>
If the data set contains URLs (e.g. linking to cross-references like
PubMed),
the name of the respective column has to begin with a "@" symbol.<br>
In order to associate the gene expression data with the array CGH data
the gene annotation file is needed. In addition the annotation file has
to have a column called "ChromosomeNumber" containing the number of the
chromosome and a column called "NucleotidePosition" containing the
starting position of the ORF of the gene. Additional
columns can contain any type of continuous or categorical variable with
the variable names in the first row. </p>
<br>
<br>
Example: <br>
<br>
<pre>geneID   ChromosomeNumber  NucleotidePosition #pathway    annotation.1 ... annotation.v<br>gene.1        1              11223444        path1,path2     a.11     ...     a.1v<br>gene.2       22              NA              path1,path3     a.21     ...     a.2v<br>...<br>gene.p        X              1055            path5           a.p1     ...     a.pv<br>  </pre>
<br>
<h2>Array CGH data</h2>
<hr>
<br>
<p align="justify">
The array CGH data should have been normalized and preprocessed. A
segmentation algorithm should be applied to estimate cytogenetic gains
and losses. We used the R package <a
 href="http://www.bioconductor.org/packages/2.4/bioc/html/GLAD.html"
 target="_blank">
GLAD </a> to preprocess our example data file.<br>
The first column contains the cloneIDs of the CGH clones. To link the
states of the clones with the samples, a column with the name
"samplename.States" is needed for each sample. These columns contain
the result of the segmentation algorithm and the entries are "-1","0"
or "1", which correspond to a cytogenetic loss, no change or gain.<br>
To link the CGH data with the gene expression data, the following six
columns are needed: "ChromosomeNumber", "ChrStart", "ChrEnd",
"ChromosomeCen", "Mapping". The variables "ChrStart" and
"ChrEnd" contain the start and end nucleotide positions of each CGH
clone. "Mapping" gives the cytoband
location of the CGH clone and "ChromosomeCen" gives the nucleotide
position of the centromere of the chromosome (so it is the same for all
clones on the same chromosome). See the example for details:</p>
<br>
<br>
<pre>CloneID ChromosomeNumber ChrStart ChrEnd Mapping ChromosomeCen sample.1.States ... sample.n.States<br>clone.1        1            145     605       1q41     124300000         1                    0<br>clone.2       22            11233   12034     22q13    11800000          -1                   1 <br>...<br>clone.w        X            NA      NA        Xq22     59500000          0                    -1 <br><pre>  <br><br>
</pre></pre>
</div>
</body>
</html>
