<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
<head>
  <title>SEURAT: Visual analytics for the integrative analysis of
microarray data</title>
  <meta name="Martin Sill" content="Seurat">
  <meta name="keywords"
 content="Microarrays,Integrative,Interactive,Visualization">
  <link rel="stylesheet" href="style.css" type="text/css">
</head>
<body>
<div id="inhalt">
<h1>Download and Installation</h1>
<hr> <br>
<br>
<h2>R and Java Runtime Environment</h2>
<hr> <br>
<p align="justify"> In order to run SEURAT the latest <a
 href="http://java.sun.com/javase/downloads/index.jsp" target="_blank">JRE</a>
(or JDK) and the latest <a href="http://cran.r-project.org/" target="_blank">R</a>
version need to be installed. In addition, to perform any clustering
and seriation analysis, the R packages <a
 href="http://www.rforge.net/Rserve/index.html" target="_blank">Rserve</a>,
<a href="http://cran.r-project.org/web/packages/amap/index.html">amap</a>
, <a href="http://cran.r-project.org/web/packages/seriation/index.html"
 target="_blank">seriation</a> and <a href="http://cran.r-project.org/web/packages/biclust/index.html"
 target="_blank">biclust</a> need to be installed.</p>
This can be done by typing the following into the R console:<br>
<br>
<pre>  install.packages(c("Rserve","amap","seriation","biclust"))<br>  </pre>
<br>
<h2>Download SEURAT</h2>
<hr> <br>
By downloading any version of SEURAT, you accept the following license:
<br>
<br>
<pre>This program is free software; you can redistribute it and/or modify<br>it under the terms of the GNU General Public License as published by<br>the Free Software Foundation; either version 3 of the License, or (at<br>your option) any later version.<br><br>This program is distributed in the hope that it will be useful, but<br>WITHOUT ANY WARRANTY; without even the implied warranty of<br>MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU<br>General Public License for more details.<br><br>You should have received a copy of the GNU General Public License<br>along with this program; if not, see <a
 href="http://www.gnu.org/licenses/" target="_blank">http://www.gnu.org/licenses/</a>.<br></pre>
<br>
<br>
<b>Version 0.91 September 10, 2009</b> <br>
<br>
<a href="download/SeuratMac.zip">Seurat</a>  (Application for Mac)<br>
<a href="download/SeuratWin.zip">Seurat</a>  (Windows batch and Jar file)<br>
<a href="download/Seurat.jar">Seurat.jar</a> (Jar file for Unix)<br>
<br>
<br>
<h2>Installation and Run</h2>
<hr> <br>
<br>
<p class="textos">To install just download the appropriate file, and:</p>
<ul>
  <br>
  <br>
  <li><strong>Windows</strong>: extract the zip archive and double click the Seurat.bat or type Seurat.bat in the command line. <br>
    <br>
  </li>
  <li><strong>Linux</strong>: launch the Seurat.jar file via the
terminal by: <br>
    <br>
    <pre>java -jar Seurat.jar</pre>
  </li>
  <br>
  <li><strong>MacOS X</strong>: extract the zip archive and double click the Seurat.app in the Seurat folder<br>
  </li>
  <br>
  <br>
</ul>
<p align="justify"> If SEURAT returns an "calculation failed" error
message after running any clustering or seriation algorithm, the
software could not connect to Rserve. In this case you have to lauch
Rserve manually by typing the following into the R console: </p>
<br>
<br>
<pre>
library(Rserve)
library(amap)
library(seriation)
library(biclust)
Rserve()
</pre>
<br>
<h2>Sample Data</h2>
<hr> <br>
Here are some sample data sets, which are ready to load and test with
SEURAT:
<br>
<br>
<code> GeneExpression.txt</code>&nbsp;&nbsp;&nbsp; gene expression
values of 7.705 genes of 50 samples <br>
<br>
<code> ClinicalData.txt </code>&nbsp;&nbsp;&nbsp; 16 clinicial
variables collected from the 50 samples
<br>
<br>
<code> GeneAnnotations.txt </code>&nbsp;&nbsp;&nbsp; gene annotations
of the 7.705 genes
<br>
<br>
<code> aCGHData.txt </code>&nbsp;&nbsp;&nbsp; CGH data for 7.873 clones
as measured by the 50 samples
<br>
<br>
The mapping of the 7.873 CGH clones to the 7.705 genes is time consuming,
thus loading the array-CGH file will last some time (depending on your
machine).
<br>
<br>
<a href="download/SampleData.zip">SampleDataI.zip</a>
<br>
<br>
Most of the seriation algorithms (e.g. ARSA, BBURCG, BBWRCG, TSP, Chen)
will not run with data sets of dimension 7000.
To test the seriation algorithms, please try the short version of the
data files: <br>
<br>
<code> GeneExpression.txt</code>&nbsp;&nbsp;&nbsp; gene expression
values of 299 genes of 50 samples <br>
<br>
<code> ClinicalData.txt </code>&nbsp;&nbsp;&nbsp; 16 clinicial
variables collected from the 50 samples
<br>
<br>
<code> GeneAnnotations.txt </code>&nbsp;&nbsp;&nbsp; gene annotations
of the 299 genes
<br>
<br>
<code> aCGHData.txt </code>&nbsp;&nbsp;&nbsp; CGH data for 1.144 clones
(chromosome 1) as measured by the 50 samples
<br>
<br>
<a href="download/SampleDataShort.zip">SampleDataShort.zip</a> 
<br>
<br>
The second example data set is based on Affymetrix exon (GeneChip Human Exon 1.0 ST Arrays) and SNP arrays (Genome-Wide Human SNP Arrays 6.0)
Most of the preprocessing of the Affymetrix chips has been done using the R-package <a href="http://cran.r-project.org/web/packages/aroma.affymetrix/index.html" target="_blank"> aroma.affymetrix</a>.
Help regarding this package and the implemented statistical methods can be found <a href="http://groups.google.com/group/aroma-affymetrix/" target="_blank">here</a>.   
In addition we provide a R-script that describes the preprocessing of the raw data: 
<br>
<br>
<code> GeneExpression.txt</code>&nbsp;&nbsp;&nbsp; gene expression values of 17.502 genes of 8 samples <br>
<br>
<code> ClinicalData.txt </code>&nbsp;&nbsp;&nbsp; 16 clinicial variables collected from the 8 samples
<br>
<br>
<code> GeneAnnotations.txt </code>&nbsp;&nbsp;&nbsp; gene annotations of the 17.502 genes
<br>
<br>
<code> SNPdata.txt </code>&nbsp;&nbsp;&nbsp; 198.965 SNPs measured by the 8 samples
<br>
<br>
<a href="download/SampleDataAffy.zip">SampleDataAffy.zip</a> 
<br>
<br>
<a href="download/preprocessing.R" >preprocessing.R</a></div>
<br>
<br>
</body>
</html>
