<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<html>
  <head>
    <title>
      Seurat: Visual analytics for the integrative analysis of microarray data
    </title>
 	<link rel="stylesheet" href="style.css" type="text/css"> 
 
 
  </head>
  <body>
  <div id="seite">
  
  <div id="kopfbereich">
  <span style="font-weight: bold">Seurat:</span>
  <span> Visual analytics for the integrative analysis of microarray data </span>
  </div>
  <hr/>
  <div id="steuerung">
  <a href="#Introduction">Introduction</a><br/><br/>
  <a href="#Features">Features</a><br/><br/>
  <a href="#Download">Download</a></a><br/><br/>
  <a href="#Video">Video</a><br/><br/>
  </div>
  <div id="inhalt">
  <a name="Introduction"></a><h2>Introduction</h2>
<p> <img src="images/seurat.png" width=500 px align="right">
	The Seurat software tool allows to perform interactive analysis of complex integrative datasets.
	At present, Seurat can integrate gene expression data, gene annotations, clinical data and data arising from array CGH.
	To keep track of the different data sets loaded into the workspace, Seurat provides a data manager,
   which organizes the data in different objects, e.g. genes, samples, CGH clones and chromosomes.
   The manager allows to browse through data sets and objects and to access the gene annotations and clinical data.
   The gene expression matrix is visualized by a heatmap, where the gene expression levels are represented by colors. 
   This completely interactive heatmap occupies a central position in Seurat and therefore is the starting point for exploratory analysis.
   To reorder the gene expression matrix, the user can choose from different clustering and seriation algorithms.
   Currently, Seurat provides agglomerative hierarchical clustering and kmeans partitioning, but in general any clustering method available in R can be used.
   Clustering results are visualized within a new interactive heatmap display and in case of hierarchical clustering together with an interactive dendrogram. 
   Results of kmeans clusterings can be compared by a confusion matrix plot.
   This graphical tool displays objects falling into the same clusters by squares with sizes proportional to the number of objects.
   The clinical data as well as gene annotations can be selected from the data manager. 
   In this context the manager automatically distinguishes between categorical and continuous data.
   Depending on the scale of the data, associated information is presented in either barcharts or histograms.
   Pre-processed array CGH data holding information about cytogenetic gains and losses can be explored with the chromosome browser.
   The chromosome browser provides a global view on all chromosomes of the human genome.
   The relative frequencies of gains and losses are visualized by histograms along the chromosomes.
   In addition each chromosome can be explored in single display.
   All of the described graphics are linked with each other, so any transient selection of objects in a graphical display will result in highlighting of the according objects in all other associated graphics. 
   In the heatmap menu the color settings of the heatmap can be changed.
   In addition it is possible to make selections persistent by assigning colors.
</p>
	<a name="Features"></a><h2>Features</h2>
	<a name="Download"></a><h2>Download</h2>
	<a name="Video"></a><h2>Video</h2>
	a video will come 
</div>
		
<div id="fussbereich">
  Contact:<a href="mailto:m.sill@dkfz.de">m.sill@dkfz.de</a>
  </div>

  </body>
</html>

